### PR TITLE
fix(logger): use one ticker for all time-related operations

### PR DESCRIPTION
**Pull request by bacongobbler**
In a separate goroutine, we were sleeping for the exact same time as
publishInterval was set to. As @kalbasit mentioned in [a comment][1],
it's better to use a ticker anyways. This removes one extra underlying
system call to clock_gettime() by making both publishKeys() and the
log drain poll share the same ticker event.

refs #4030, as I found this using the .scap dump. I don't think this will
drop the CPU usage significantly, but it's a start.

[1]: https://github.com/deis/deis/pull/3795#discussion_r31878237

### CODE REVIEW

**Comment by technosophos** [comment]
Code LGTM

**Comment by mboersma** [comment]
Code LGTM.

`Showing  1 changed file  with 12 additions and 29 deletions.` :smile:

File: logger/main.go [hunk 1]
```diff
@@ -76,12 +80,6 @@ func main() {
...
 			}
-			time.Sleep(time.Duration(publishInterval) * time.Second)
-		}
-	}()
-
-	for {
-		select {
 		case <-signalChan:
 			close(exitChan)
 			stopChan <- true
```
Comment (kalbasit) [inline logger/main.go]: You should also stop the ticker. Github doesn't allow me to add a comment on a line that is hidden.

```
 		case <-cleanupChan:
 			return
```

To

```
 		case <-cleanupChan:
 			t.Stop()
 			return
```

to stop the ticket (avoid leaking goroutines)

File: logger/main.go [hunk 2]
```diff
@@ -97,21 +95,6 @@ func publishKeys(client *etcd.Client, host, etcdPath, port string, ttl uint64) {
...
-// something on exitChan.
-func publishService(exitChan chan bool, client *etcd.Client, host string, etcdPath string, port string, ttl uint64) {
-	publishKeys(client, host, etcdPath, port, ttl)
-	t := time.NewTicker(time.Duration(publishInterval) * time.Second)
-	for {
-		select {
-		case <-t.C:
-			publishKeys(client, host, etcdPath, port, ttl)
-		case <-exitChan:
-			return
```
Comment (kalbasit) [inline logger/main.go]: probably my mistake here I forgot to stop the ticker :(

File: logger/main.go (hunk 1 above)
Comment (bacongobbler) [inline logger/main.go]: good catch. Thanks!

**Comment by bacongobbler** [comment]
done. now it's calling `ticker.Stop()` on `cleanupChan` as well as I removed an unused variable (`stopChan`) which was probably from an older refactor.

**Comment by mboersma** [comment]
LGTM++.

**Pull request merged**